src/OAuth2/Encryption/EncryptionInterface.php

```diff
@@ -4,8 +4,9 @@
 
 interface EncryptionInterface
 {
+    public function getSupportedAlgorithms($type = null);
     public function encode($payload, $key, $algorithm = null);
-    public function decode($payload, $key, $algorithm = null);
+    public function decode($payload, $key, $allowed_algorithms = array());
```

Comment:
I would prefer to keep this `null` to preserve backwards compatibility, and we can change it to `array()` in the next major version